sql/expression/function/aggregation/window_framer.og.go

```diff
@@ -0,0 +1,673 @@
+	startCurrentRow := true
+	endFollowingExpr := frame.EndFollowingExpr()
+	return &RangeCurrentRowToFollowingExprFramer{
+		rangeFramerBase{
+			startCurrentRow:  startCurrentRow,
+			endFollowingExpr: endFollowingExpr,
+		},
+	}, nil
+}
+
+type RangeCurrentRowToUnboundedFollowingFramer struct {
+	rangeFramerBase
+}
+
+var _ sql.WindowFramer = (*RangeCurrentRowToUnboundedFollowingFramer)(nil)
+
+func NewRangeCurrentRowToUnboundedFollowingFramer(frame sql.WindowFrame) (sql.WindowFramer, error) {
+	startCurrentRow := true
+	unboundedFollowing := true
+	return &RangeCurrentRowToUnboundedFollowingFramer{
+		rangeFramerBase{
+			startCurrentRow:    startCurrentRow,
+			unboundedFollowing: unboundedFollowing,
+		},
+	}, nil
+}
+
+type RangeFollowingExprToPrecedingExprFramer struct {
+	rangeFramerBase
+}
+
+var _ sql.WindowFramer = (*RangeFollowingExprToPrecedingExprFramer)(nil)
+
+func NewRangeFollowingExprToPrecedingExprFramer(frame sql.WindowFrame) (sql.WindowFramer, error) {
+	startFollowingExpr := frame.StartFollowingExpr()
+	endPrecedingExpr := frame.EndPrecedingExpr()
+	return &RangeFollowingExprToPrecedingExprFramer{
+		rangeFramerBase{
+			startFollowingExpr: startFollowingExpr,
+			endPrecedingExpr:   endPrecedingExpr,
+		},
+	}, nil
+}
+
+type RangeFollowingExprToCurrentRowFramer struct {
+	rangeFramerBase
+}
+
+var _ sql.WindowFramer = (*RangeFollowingExprToCurrentRowFramer)(nil)
+
+func NewRangeFollowingExprToCurrentRowFramer(frame sql.WindowFrame) (sql.WindowFramer, error) {
+	startFollowingExpr := frame.StartFollowingExpr()
+	endCurrentRow := true
+	return &RangeFollowingExprToCurrentRowFramer{
+		rangeFramerBase{
+			startFollowingExpr: startFollowingExpr,
+			endCurrentRow:      endCurrentRow,
+		},
+	}, nil
+}
+
+type RangeFollowingExprToFollowingExprFramer struct {
+	rangeFramerBase
+}
+
+var _ sql.WindowFramer = (*RangeFollowingExprToFollowingExprFramer)(nil)
+
+func NewRangeFollowingExprToFollowingExprFramer(frame sql.WindowFrame) (sql.WindowFramer, error) {
+	startFollowingExpr := frame.StartFollowingExpr()
+	endFollowingExpr := frame.EndFollowingExpr()
+	return &RangeFollowingExprToFollowingExprFramer{
+		rangeFramerBase{
+			startFollowingExpr: startFollowingExpr,
+			endFollowingExpr:   endFollowingExpr,
+		},
+	}, nil
+}
+
+type RangeFollowingExprToUnboundedFollowingFramer struct {
+	rangeFramerBase
+}
+
+var _ sql.WindowFramer = (*RangeFollowingExprToUnboundedFollowingFramer)(nil)
+
+func NewRangeFollowingExprToUnboundedFollowingFramer(frame sql.WindowFrame) (sql.WindowFramer, error) {
+	startFollowingExpr := frame.StartFollowingExpr()
+	unboundedFollowing := true
+	return &RangeFollowingExprToUnboundedFollowingFramer{
+		rangeFramerBase{
+			startFollowingExpr: startFollowingExpr,
+			unboundedFollowing: unboundedFollowing,
+		},
+	}, nil
+}
+
+func NewFramer(frame sql.WindowFrame) (sql.WindowFramer, error) {
+	switch frame.Op() {
+	case 0:
```

Comment:
Better to generate each of these framers their own NewFramer method that you call